pkg/sql/conn_executor.go

```diff
@@ -2792,6 +2792,7 @@ func (ex *connExecutor) initEvalCtx(ctx context.Context, evalCtx *extendedEvalCo
 		Context: eval.Context{
 			Planner:                        p,
 			StreamManagerFactory:           p,
+			SystemJobsIteratorFactory:      p,
 			PrivilegedAccessor:             p,
 			SessionAccessor:                p,
 			JobExecContext:                 p,
```

Comment:
there is quite a lot of changed code related to this context business...
I'm not 100% clear on why we need it (but seeing other examples, i.e. stream manager factory), we probably do
need this.  However, given that we are iterating over system jobs, have you considered adding required functionality to JobExecContext? I think it might be appropriate (i.e. we are iterating over system jobs), and 
planner already implements JobExecContext...